src/caret_analyze/record/record.py

```diff
@@ -793,6 +793,25 @@ def groupby(self, columns: list[str]) -> dict[tuple[int, ...], RecordsInterface]
 
         return group
 
+    def check_null(self) -> bool:
```

Comment:
The function name `check_null` doesn't actually perform a null check, so there seems to be a mismatch between the name and the actual functionality. It might be more appropriate to use a name like `has_mismatched_columns`.